test/dialects/soql_test.py

```diff
@@ -2,22 +2,41 @@
 import pytest
 
 from sqlfluff.core import FluffConfig, Linter
+from sqlfluff.core.errors import SQLParseError
 
 
 @pytest.mark.parametrize(
     "raw",
     [
-        "ALTER TABLE foo DROP COLUMN bar",
-        "CREATE USER my_user",
-        "TRUNCATE TABLE foo",
-        "EXPLAIN SELECT Id FROM Contact",
-        "DROP TABLE foo",
-        "DROP USER my_user",
+        """ALTER TABLE foo DROP COLUMN bar
+        """,
+        """CREATE USER my_user
+        """,
+        """TRUNCATE TABLE foo
+        """,
+        """EXPLAIN SELECT Id FROM Contact
+        """,
+        """DROP TABLE foo
+        """,
+        """DROP USER my_user
+        """,
     ],
 )
 def test_non_selects_unparseable(raw: str) -> None:
     """Test that non-SELECT commands are not parseable."""
     cfg = FluffConfig(configs={"core": {"dialect": "soql"}})
     lnt = Linter(config=cfg)
     result = lnt.lint_string(raw)
-    assert result.num_violations() > 0
+    assert len(result.violations) == 1
+    assert isinstance(result.violations[0], SQLParseError)
+
+
+def test_ignore_unreferenced_object() -> None:
```

Comment:
This test should just be added to the L026.yml file rather than as a Python test.

The rule YAML files can’t check as much as pure Python (e.g. they can’t check for unparsable errors) but for simple parsing, rule passes or rule fixes we try to use them as simpler.

Reply:
Yeah, I just discovered this file from looking at other PRs. Will do. 👍